massmailer/query_parser.py

```diff
@@ -380,3 +392,25 @@ def parse(tokens):
         ).setParseAction(parse)
 
         return stmt.parseString(query)[0]
+
+
+REGISTERED_ENUMS = {}
+
+
+def register_enum(enum_name):
```

Comment:
`register_enum(namespace, enum_name=None)`
and auto-populate `enum_name` from `cls.__name__` if `enum_name` is not None.